packages/providers/src.ts/hedera-provider.ts

```diff
@@ -0,0 +1,130 @@
+ * As every provider, this one also gives us read-only access.
+ *
+ * Constructable with a string, which automatically resolves to a hedera network via the hashgraph SDK.
+ */
+export class DefaultHederaProvider extends BaseProvider {
+    private readonly hederaClient: Client;
+    private readonly hederaNetwork: string;
+
+    constructor(network: string) {
+        super('testnet');
+        this.hederaNetwork = network;
+        this.hederaClient = Client.forName(resolveNetwork(network))
+    }
+
+    /**
+     *  AccountBalance query implementation, using the hashgraph sdk.
+     *  It returns the tinybar balance of the given address.
+     *
+     * @param addressOrName The address to check balance of
+     * @param blockTag -  not used. Will throw if used.
+     */
+    async getBalance(addressOrName: string | Promise<string>, blockTag?: BlockTag | Promise<BlockTag>): Promise<BigNumber> {
+        if (blockTag || await blockTag) {
+            logger.throwArgumentError("Cannot use blockTag for hedera services.", "blockTag", blockTag);
+            return BigNumber.from(0);
+        }
+        addressOrName = await addressOrName;
+        const { shard, realm, num } = getAccountFromAddress(addressOrName);
+        const shardNum = BigNumber.from(shard).toNumber();
+        const realmNum = BigNumber.from(realm).toNumber();
+        const accountNum = BigNumber.from(num).toNumber();
+        const balance = await new AccountBalanceQuery()
+            .setAccountId(new AccountId({ shard: shardNum, realm: realmNum, num: accountNum }))
+            .execute(this.hederaClient);
+        return BigNumber.from(balance.hbars.toTinybars().toNumber());
+    }
+
+    /**
+     * Transaction record query implementation using the mirror node REST API.
+     *
+     * @param txId - id of the transaction to search for
+     */
+    async getTransaction(txId: string | Promise<string>): Promise<TransactionResponse> {
+        txId = await txId;
+        const ep = '/api/v1/transactions';
+        const url = resolveMirrorNetworkUrl(this.hederaNetwork);
+        // The following logic will be refactored/moved to tx.wait() method
+        const maxRetries = 10;
+        let counter = 0;
+        while (true) {
+            if (counter >= maxRetries) {
+                logger.warn(`Giving up after ${maxRetries} retries.`);
+                return null;
+            }
+            const { data } = await axios.get(url + ep);
+            const filtered = data.transactions
+                .filter((e: { transaction_id: string | Promise<string>; }) => e.transaction_id === txId);
+            if (filtered.length > 0) {
+                return filtered[0];
+            }
+            // retry each 0.5 seconds
+            await sleep(500);
+            counter++;
+        }
+    }
+
+    /**
+     * Allows us to get the underlying gRPC client and execute gRPC calls.
+     */
+    public getClient(): Client {
```

Comment:
We should not expose this. Everything should be wrapped into the `hedera-provider`